torch/_inductor/comms.py

```diff
@@ -203,6 +206,103 @@ def _temp_group_visit_leaves(snode, fn):
         fn(snode)
 
 
+def wait_exposed_communication_time(
+    snodes_to_wait, runtimes
+) -> tuple[float, float, str]:
```

Comment:
Woudl you mind adding type annotations, and a docstring to all of these functions ? it makes a lot easier to read/review